src/main/steps/home/get.ts

```diff
@@ -28,40 +30,64 @@ export class HomeGetController {
     );
     const isFirstQuestionComplete = firstQuestionForm.getErrors(req.session.userCase).length === 0;
 
-    if (req.session.userCase.state === State.Holding) {
-      res.redirect(HUB_1);
-    }
-
     if (req.session.isApplicant2) {
-      return res.redirect(
-        isFirstQuestionComplete
-          ? `${APPLICANT_2}${CHECK_ANSWERS_URL}`
-          : `${APPLICANT_2}${YOU_NEED_TO_REVIEW_YOUR_APPLICATION}`
+      res.redirect(applicant2RedirectPageSwitch(req.session.userCase.state, isFirstQuestionComplete));
+    } else {
+      res.redirect(
+        applicant1RedirectPageSwitch(req.session.userCase.state, req.session.userCase, isFirstQuestionComplete)
       );
     }
+  }
+}
+
+const applicant1RedirectPageSwitch = (caseState: State, userCase: Partial<Case>, isFirstQuestionComplete: boolean) => {
+  let redirectLink;
 
-    switch (req.session.userCase.state) {
-      case State.AwaitingApplicant1Response: {
-        req.session.userCase.applicant2ScreenHasUnionBroken === YesOrNo.NO
-          ? res.redirect(APPLICATION_ENDED)
-          : res.redirect(CHECK_ANSWERS_URL);
-        break;
-      }
-      case State.AwaitingApplicant2Response: {
-        res.redirect(SENT_TO_APPLICANT2_FOR_REVIEW);
-        break;
-      }
-      case State.Applicant2Approved: {
-        res.redirect(CONFIRM_JOINT_APPLICATION);
-        break;
-      }
-      case State.Submitted: {
-        res.redirect(APPLICATION_SUBMITTED);
-        break;
-      }
-      default: {
-        res.redirect(isFirstQuestionComplete ? CHECK_ANSWERS_URL : YOUR_DETAILS_URL);
-      }
+  switch (caseState) {
+    case State.AwaitingApplicant1Response: {
+      redirectLink = userCase.applicant2ScreenHasUnionBroken === YesOrNo.NO ? APPLICATION_ENDED : CHECK_ANSWERS_URL;
+      break;
+    }
+    case State.AwaitingApplicant2Response: {
+      redirectLink = SENT_TO_APPLICANT2_FOR_REVIEW;
+      break;
+    }
+    case State.Applicant2Approved: {
+      redirectLink = CONFIRM_JOINT_APPLICATION;
+      break;
+    }
+    case State.Submitted: {
+      redirectLink = APPLICATION_SUBMITTED;
+      break;
+    }
+    case State.Holding: {
+      redirectLink = HUB_1;
+      break;
+    }
+    default: {
+      redirectLink = isFirstQuestionComplete ? CHECK_ANSWERS_URL : YOUR_DETAILS_URL;
     }
   }
-}
+  return redirectLink;
```

Comment:
might be better to just return in the switch case

Reply:
also applies for the other method. this way you dont need a variable set, just return the url in each case and default